### PR TITLE
Update README and introduce Developer instructions

### PR DESCRIPTION
**Pull request by jim-portegies**

### CODE REVIEW

File: Developer-instructions.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+brew install opam
+```
+or macports
+```
+port install opam
+```
+
+### Creating the desired opam environment: simplified version
+When we develop, we need to be aware of the version of Coq we develop for.
+Most development is done against the main branch, and then the following setup will suffice.
```
Comment (pimotte) [inline Developer-instructions.md:29]: Maybe
```suggestion
Most development is done based off the main branch, and then the following setup will suffice.
```
"against" made me think it was the main branch of something else (oCaml or Coq)

File: Developer-instructions.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+```
+opam init
+eval $(opam env)
+opam install coq-lsp.2.0.0+8.17
```
Comment (pimotte) [inline Developer-instructions.md:34]: How often does this version change? If it changes often maybe note that above this command, if there's a way we can reliably install some latest version, maybe give that command instead. Maybe the edit below is enough?

File: Developer-instructions.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+opam install ocaml-lsp-server
+```
+
```
Comment (pimotte) [inline Developer-instructions.md:37]:
```suggestion
replacing 8.17 with the desired version of Coq

```

File: Developer-instructions.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+Regardless of how you initialized opam, you can now add a new switch with
+```
+opam switch create your_preferred_switch_name ocaml-base-compiler.4.14.1
+```
+Next you can install the background libraries again
+```
+opam install coq-lsp.2.0.0+8.17
+opam install ocaml-lsp-server
+```
+replacing 8.17 with the desired version of Coq
```
Comment (pimotte) [inline Developer-instructions.md:55]:
```suggestion
again replacing 8.17 with the desired version of Coq
```

File: Developer-instructions.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+```
+dune install -p coq-waterproof
+```
+
+## Testing
+
+We aim to test every feature that we implement. Currently we test almost
+all functionality present in `.v` files. Note for instance that tactics
+such as `Choose`, `Assume`, both have a file that defines them, and a file
+that tests them.
```
Comment (pimotte) [inline Developer-instructions.md:92]: How do I run the tests?

File: Developer-instructions.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+autoreconf -i -s
+./configure
+```
+after which one could run
+```
+make
+make install
+```
+
+One needs to have the `autoconf` ocaml libary installed. One can install this with
```
Comment (pimotte) [inline Developer-instructions.md:137]: Or equivalent commands for non-Debian environments.

File: Developer-instructions.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+Hence, we can convert `f` into something that outputs a tactic by
+
+```
+let my_f_tactic : a -> b -> c -> d tactic =
+  fun input_a input_b input_c -> tclUNIT (f input_a input_b input_c)
+```
+
+### Make a `valexpr` tactic
+
+Yet datatypes in Ocaml and in Ltac2 need to be translated to each other
```
Comment (pimotte) [inline Developer-instructions.md:174]:
```suggestion
Datatypes in Ocaml and in Ltac2 need to be translated to each other
```
If this was into contrast to the part before I don't really understand it, so alternative to removing yet is explaining the contrast in a bit more depth.

File: Developer-instructions.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+by using an intermediate datatype: namely `valexpr`. This also means that
+not any tactic can pass the ffi, but only `valexpr`-valued tactics.
```
Comment (pimotte) [inline Developer-instructions.md:176]:
```suggestion
by using an intermediate datatype: namely `valexpr`. This also means that
only `valexpr`-valued tactics can pass the ffi.
```
For clarity?

File: Developer-instructions.md [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+make
+make install
+```
+
+One needs to have the `autoconf` ocaml libary installed. One can install this with
+```
+sudo apt-get update
+sudo apt-get install autoconf
+```
+
```
Comment (pimotte) [inline Developer-instructions.md:142]:
```suggestion
or equivalent for non-Debian/Ubuntu based environments

```

**Review by pimotte — APPROVED** [review]
Couple small suggestions, other than that, looks good to me:)

File: Developer-instructions.md (hunk 2 above)
Comment (jim-portegies) [reply Developer-instructions.md:34]: I'll go with your suggestion below. We may indeed want to think about how to get all version information up to date.

File: Developer-instructions.md (hunk 5 above)
Comment (jim-portegies) [reply Developer-instructions.md:93]: I'll add an explanation

File: Developer-instructions.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+```
+opam init --bare
+``` 
+Regardless of how you initialized opam, you can now add a new switch with
+```
+opam switch create your_preferred_switch_name ocaml-base-compiler.4.14.1
+```
+Next you can install the background libraries again
+```
+opam install coq-lsp.2.0.0+8.17
```
Comment (ejgallego) [inline Developer-instructions.md:54]: Think you folks got a typo here, do you mean `coq-lsp.0.2.0+8.17` ?

Comment (jim-portegies) [reply Developer-instructions.md:54]: Oops! You're right, thanks!

**Pull request merged**